src/main/java/com/synopsys/integration/alert/component/authentication/actions/AuthenticationTestAction.java

```diff
@@ -0,0 +1,141 @@
+        }
+
+        if (samlEnabled) {
+            performSAMLTest(registeredFieldValues);
+        }
+
+        return new MessageResult("Successfully tested authentication configuration.");
+    }
+
+    private void performLdapTest(FieldModel fieldModel, FieldAccessor registeredFieldValues) throws IntegrationException {
+        logger.info("LDAP enabled testing LDAP authentication.");
+        String userName = fieldModel.getFieldValue(AuthenticationUIConfig.TEST_FIELD_KEY_USERNAME).orElse("");
+        Optional<LdapAuthenticationProvider> ldapProvider = ldapManager.createAuthProvider(registeredFieldValues);
+        String errorMessage = String.format("Ldap Authentication test failed for the test user %s.  Please check the LDAP configuration.", userName);
+        Map<String, String> errorsMap = new HashMap<>();
+        if (!ldapProvider.isPresent()) {
+            errorsMap.put(AuthenticationDescriptor.KEY_LDAP_ENABLED, errorMessage);
+        } else {
+            Authentication pendingAuthentication = new UsernamePasswordAuthenticationToken(userName,
+                fieldModel.getFieldValue(AuthenticationUIConfig.TEST_FIELD_KEY_PASSWORD).orElse(""));
+            Authentication authentication = ldapProvider.get().authenticate(pendingAuthentication);
+            if (!authentication.isAuthenticated()) {
+                errorsMap.put(AuthenticationDescriptor.KEY_LDAP_ENABLED, errorMessage);
+            }
+        }
+
+        if (!errorsMap.isEmpty()) {
+            throw new AlertFieldException(errorsMap);
+        }
+    }
+
+    private void performSAMLTest(FieldAccessor registeredFieldValues) throws IntegrationException {
+        Optional<ConfigurationFieldModel> metaDataURLField = registeredFieldValues.getField(AuthenticationDescriptor.KEY_SAML_METADATA_URL);
+        Optional<ConfigurationFieldModel> metaDataFileField = registeredFieldValues.getField(AuthenticationDescriptor.KEY_SAML_METADATA_FILE);
+        boolean testMetaDataURL = metaDataURLField.map(ConfigurationFieldModel::isSet).orElse(false);
+        boolean testMetaDataFile = metaDataFileField.map(ConfigurationFieldModel::isSet).orElse(false);
+        Map<String, String> errorsMap = new HashMap<>();
+        if (testMetaDataURL) {
+            logger.info("Testing SAML Metadata URL...");
+            try {
+                Optional<ExtendedMetadataDelegate> provider = samlManager.createHttpProvider(registeredFieldValues.getStringOrEmpty(AuthenticationDescriptor.KEY_SAML_METADATA_URL));
+                if (provider.isPresent()) {
+                    provider.get().initialize();
+                }
+            } catch (MetadataProviderException ex) {
+                logger.error("Testing SAML Metadata URL error: ", ex);
+                errorsMap.put(AuthenticationDescriptor.KEY_SAML_METADATA_URL, ex.getMessage());
+            } catch (Exception ex) {
+                logger.error("Unexpected error testing SAML:", ex);
```

Comment:
Should we put something in the errorMap here saying an unexpected error occurred?

Reply:
To be safe I will just catch the exception and add it to the map.